controllers/drplacementcontrol.go

```diff
@@ -462,6 +456,206 @@ func (d *DRPCInstance) getCurrentHomeClusterName(toCluster string, drClusters []
+// Returns:
+//   - bool: Indicating if prerequisites are met
+//   - error: Any error in determining the prerequisite status
+func (d *DRPCInstance) checkFailoverPrerequisites(failoverCluster string) (bool, error) {
+	var (
+		met bool
+		err error
+	)
+
+	if isMetroAction(d.drPolicy, d.drClusters, failoverCluster, d.instance.Spec.FailoverCluster) {
+		met, err = d.checkMetroFailoverPrerequisites(failoverCluster)
+	} else {
+		met, err = d.checkRegionalFailoverPrerequisites(failoverCluster)
+	}
+
+	if err == nil && met {
+		return true, nil
+	}
+
+	msg := "Waiting for spec.failoverCluster to meet failover prerequsites"
+
+	if err != nil {
+		msg = err.Error()
+
+		rmnutil.ReportIfNotPresent(d.reconciler.eventRecorder, d.instance, corev1.EventTypeWarning,
+			rmnutil.EventReasonSwitchFailed, err.Error())
+	}
+
+	d.setDRPCCondition(&d.instance.Status.Conditions, rmn.ConditionAvailable, d.instance.Generation,
+		d.getConditionStatusForTypeAvailable(), string(d.instance.Status.Phase), msg)
+
+	return met, err
+}
+
+// checkMetroFailoverPrerequisites checks for any MetroDR failover prerequsites that need to be met on the
+// failoverCluster before initiating a failover.
+// Returns:
+//   - bool: Indicating if prerequisites are met
+//   - error: Any error in determining the prerequisite status
+func (d *DRPCInstance) checkMetroFailoverPrerequisites(failoverCluster string) (bool, error) {
+	met := true
+
+	d.setProgression(rmn.ProgressionWaitForFencing)
+
+	fenced, err := d.checkClusterFenced(failoverCluster, d.drClusters)
+	if err != nil {
+		return !met, err
+	}
+
+	if !fenced {
+		return !met, fmt.Errorf("current home cluster %s is not fenced", failoverCluster)
+	}
+
+	return met, nil
+}
+
+// checkRegionalFailoverPrerequisites checks for any RegionalDR failover prerequsites that need to be met on the
+// failoverCluster before initiating a failover.
+// Returns:
+//   - bool: Indicating if prerequisites are met
+//   - error: Any error in determining the prerequisite status
+func (d *DRPCInstance) checkRegionalFailoverPrerequisites(failoverCluster string) (bool, error) {
+	d.setProgression(rmn.ProgressionWaitForStorageMaintenenceActivation)
+
+	if required, activationsRequired := d.requiresRegionalFailoverPrequisites(failoverCluster); required {
+		return d.checkClusterMaintenanceActivation(failoverCluster, activationsRequired)
+	}
+
+	return true, nil
+}
+
+// requiresRegionalFailoverPrequisites checks protected PVCs as reported by the last known Primary cluster
+// to determine if this instance requires failover maintenance modes to be active prior to initiating
+// a failover
+func (d *DRPCInstance) requiresRegionalFailoverPrequisites(failoverCluster string) (
+	bool,
+	map[string]rmn.StorageIdentifiers,
+) {
+	activationsRequired := map[string]rmn.StorageIdentifiers{}
+
+	vrg := d.getLastKnownPrimaryVRG(failoverCluster)
+	if vrg == nil {
+		// TODO: Is this an error, should we ensure at least one VRG is found in the edge cases?
```

Comment:
On Failover, most likely the last known primary is unreachable, hence, the vrg will equal nil. Maybe on failover we should always get it from the S3Store as we discussed.

Reply:
This is a pending TODO, will raise an issue and fix it subsequently.

Reply:
Tracked in #835